Add MambaSSM as a library

Wauplin opened the pull request:
This PR adds https://github.com/state-spaces/mamba as an official library on the Hub.
Let's wait for Mamba integration to be merged (coming soon) + having a few first models uploaded on the Hub.

cc @osanseviero @NielsRogge @tridao

Related PR on Mamba side: https://github.com/state-spaces/mamba/pull/471

osanseviero commented on packages/tasks/src/model-libraries.ts:
```diff
@@ -298,6 +298,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
+		repoName: "MambaSSM",
+		repoUrl: "https://github.com/state-spaces/mamba",
+		filter: false,
+		countDownloads: `path:"config.json"`,
```
No needed as this is already one of the files used to count downloads

Wauplin replied:
addressed in https://github.com/huggingface/huggingface.js/pull/802/commits/efaf43986ac117fb048848d30ea705f706e60aac

osanseviero reviewed (approved):
LGTM if naming is agreed

julien-c commented on packages/tasks/src/model-libraries.ts:
```diff
@@ -298,6 +298,13 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		repoName: "mindspore",
 		repoUrl: "https://github.com/mindspore-ai/mindspore",
 	},
+	mamba_ssm: {
```
just `mamba` would be cleaner, branding-wise?

```suggestion
	mamba: {
```

julien-c replied:
and it's `pip install mamba-ssm` on PyPI so maybe a better option too

I think i'd go with `mamba` personally

Wauplin replied:
We discussed this with @osanseviero. The problem of simply `mamba` is that it's a `transformers` architecture as well so it might create some conflicts. Also having `mamba_ssm` as library_name and `mamba` as tag helps with code snippets (to import `Mamba` class).

Wauplin replied:
We could have `Mamba` as prettyName but it might be misleading

julien-c replied:
ok but then dash rather than underscore no?

julien-c replied:
nvm let's not complicate stuff, LGTM as is :)

Wauplin replied:
good point, addressed in https://github.com/huggingface/huggingface.js/pull/802/commits/e3e7e62c2bd3b0309e1e3a9a203abc430e17c98c and https://github.com/state-spaces/mamba/pull/471/commits/961eccb89f92e34459ea3e6959081c833bba19c3

Wauplin commented:
@osanseviero @julien-c I've updated the code snippet. In the end only `MambaLMHeadModel` is loadable from the Hub since it's a model. `Mamba`, `Mamba2` and `Mamba2Simple` are simply layers. I've also updated https://github.com/state-spaces/mamba/pull/471 to reflect that.

osanseviero reviewed (approved):
Nice :fire:

Wauplin commented:
(I would have waited for https://github.com/state-spaces/mamba/pull/471 to be merged but no big deal)
